src/Diagnostics/FxCop/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -186,7 +190,9 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
 
             private bool IsSerializable(ITypeSymbol namedTypeSymbol)
             {
-                return namedTypeSymbol.GetAttributes().Any(a => a.AttributeClass == _serializableAttributeTypeSymbol);
+                return namedTypeSymbol.IsValueType ||
+                    namedTypeSymbol.Equals(_stringSymbol) ||
```

Comment:
You can simply check if namedTypeSymbol.SpecialType == SpecialType.System_String and don't need to store the symbol anywhere.